server/src/main/java/io/crate/types/DateType.java

```diff
@@ -0,0 +1,120 @@
+/*
+ * Licensed to CRATE Technology GmbH ("Crate") under one or more contributor
+ * license agreements.  See the NOTICE file distributed with this work for
+ * additional information regarding copyright ownership.  Crate licenses
+ * this file to you under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.  You may
+ * obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.  See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ *
+ * However, if you have executed another commercial license agreement
+ * with Crate these terms will supersede the license and you may use the
+ * software solely pursuant to the terms of the relevant commercial agreement.
+ */
+
+package io.crate.types;
+
+import io.crate.Streamer;
+import org.apache.lucene.util.RamUsageEstimator;
+import org.elasticsearch.common.io.stream.StreamInput;
+import org.elasticsearch.common.io.stream.StreamOutput;
+
+import java.io.IOException;
+import java.time.Instant;
+import java.time.LocalDate;
+import java.time.ZoneOffset;
+
+public class DateType extends DataType<LocalDate>
+    implements FixedWidthType, Streamer<LocalDate> {
```

Comment:
You'll also need to add a corresponding `PGType` implementation so that values of type `DateType` can be sent via the PostgreSQL wire protocol.

See https://github.com/crate/crate/blob/0772f2d9b3d76bb8b0ad66de96257151d4ebf2c4/server/src/main/java/io/crate/protocols/postgres/types/PGTypes.java#L42

https://github.com/crate/crate/blob/1b7b2bfdd4e950068be7f11e69c48585369d0767/server/src/main/java/io/crate/protocols/postgres/types/PGType.java#L33

If we stick to using `LocalDate` as the internal type for `date` then we'll also need an implementation for how the data type is seralized to HTTP clients. See https://github.com/crate/crate/blob/master/server/src/main/java/io/crate/types/DataTypeXContentExtension.java

I wonder if we should instead stick to `Long` as internal type. That would reduce the amount of allocations on internal streaming operations.  Currently `readValueFrom` and `writeValueTo` is quite heavy.

Reply:
Could you please elaborate about having Long as internal type? I guess it means I need to replace all LocalDate usages to simple Long read/writes/casts, revert fixed size estimation to Long.size, but does it mean I will not have to change DataTypeXContentExtension.java anymore?

I will address tests & PG types & ddl doc comments accordingly to the outcome of that comment